File: Labs/Lab 6/CS/Lab6/StringClass.cs
public class StringClass : ICharacterReplacer
{
    private string content;

    public StringClass(string content)
    {
        this.content = content;
    }

    public string Content => content;

    public void ReplaceCharacter(char oldChar, char newChar)
    {
        content = content.Replace(oldChar, newChar);
    }
}

File: Labs/Lab 6/TS/StringClass.ts
import { ICharacterReplacer } from "./ICharacterReplacer";

export class StringClass implements ICharacterReplacer {
    private content: string;

    constructor(content: string) {
        this.content = content;
    }

    getContent(): string {
        return this.content;
    }

    setContent(content: string): void {
        this.content = content;
    }

    replaceCharacter(oldChar: string, newChar: string): void {
        this.content = this.content.split(oldChar).join(newChar);
    }
}
